designdocs/user-handles.md

```diff
@@ -13,55 +13,65 @@ The goal of this proposal is to enable users to create unique handles on the Fre
 * Suffixes are numeric and contain digits without any leading zeros.
-* Suffixes are u32 limited to a range defined by governance
+* Suffixes are `u16` limited to a range set as runtime `Config` parameters.
 * Suffix will be randomly constrained.
 * Homoglyph versions of handles should still resolve to the same ```msa_id``` (e.g. ```user.1234``` and ```u$er.1234``` should resolve to the same ```msa_id```).
-* After a handle is released, it cannot be reused by a time block set by governance.
+* `Display Handle` is the handle that is displayed to the user and must be unique.
+
+## Proposal
+
+Handles```registry``` on Frequency chain.
 
 ### General Steps
 
 * Create a randomly shuffled array of numeric suffixes.
 * For each base handle requested, a storage map will be created with a randomly generated start index (in to the shuffled array) and a current index.  The current index will be incremented each time a new suffix is requested.
 * When the current index reaches the start index, all suffixes are exhausted and an error is returned.
-* Query the chain to check which values are available, check if the handle with the suffix is available.
-* Choose an available value and attempt to claim it by submitting the full handle with the suffix to the chain.
+* Alternativelty: Query the chain to check which values are possi available, check if the handle with the suffix is available before claiming a handle.
 * Chain validates the handle and suffix and maps the handle to the ```msa_id```.
 * Chain maintains a reverse mapping from handle to msa for offchain use.
+* Chain provides a set of rpcs to query msa for a given `DisplayHandle` and vice versa.
+* Chain providers an rpc to get presumptive suffix for a given base handle.
 
-### Chain Steps
+### Handle replay attack
 
-* When a user submits a base handle, check the numeric suffix for availability and that it honors range defined for suffix window.
-* Store ```msa_id``` to ```handle``` mapping and ```handle``` to ```msa_id``` mapping.
+To prevent replay attacks, the chain will require a mortality period for handles. This can be included in signed payload from user as a `expiration` field. The chain will check if the `expiration` is greater than the current block number. If not, the transaction will fail.  
 
 ### Handling Race Conditions
 
-There is an exceedingly small chance that two users may attempt to claim the same handle and suffix combination within the same block. In such cases, the transaction will fail to create the MSA, and the app will receive a failure for the transaction rather than the expected MsaHandleCreated event.
+There is an exceedingly small chance that two users may attempt to claim the same handle and suffix combination within the same block. In such cases, the transaction will succeed as the distribution of suffixes by chain is random as well as on a first come basis.
 
 ## Sequence Diagram
```

Comment:
Do we want to update the sequence diagram to reflect the current state of how user handles works currently? Or at least note which parts of the diagram are not implemented yet?

Reply:
We have all parts of sequence diag

Reply:
yep, updated some terms with were out dated let me know if you find anything else